### ISSUE TITLE
namespaced rights instead of object rights

### ISSUE THREAD

**Issue by dtkahl** [dtkahl/php-access-control#10]
Currently rights are defined in 2 arrays:

```php
...
['login'],
[
    'post' => ['create', 'delete'],
    'image' => ['upload']
]
....
```

Instead defining them in one with object identifier as namespace separated by `.` would be useful:

```php
...
[
    'login',
    'post.create',
    'post.delete',
    'image.upload'
]
...
```

This brings the benefit that you can define and check object related rights without always the need of the actual object.

#### As Example

The right to create a post could not be checked on an object because it does not exist at this moment. You could define a global right `create_post` but this would result in naming inconsistancy.

With namespaced rights the check would now look like this:
```php
$judge->checkRight('post.create');
```

**Comment by dtkahl** [comment]
Definition with a bit less code:

```php
...
[
    'login',
    'post' => ['create', 'delete'],
    'image' => ['upload']
]
....

This would result in auto-namespace the rights.

**Comment by dtkahl** [comment]
This also fixes some inheritance problems because it makes the right identifier more unique.

If I check on a Post if I am allowed o Create a nested Comment something like this would not be the best solution:

```php
$judge->checkRight('create', $post)
```

I would imply the this checks if i could create a Post not a Comment, so I need to name the right `create_comment` or something similar.

Whit the change of this issue I could write:
```php
$judge->checkRight('comment.create', $post)
``

**Comment by dtkahl** [comment]
Instead of 1.1.0 this will be released with 2.0.0 because it breaks BC.
